app/Http/Middleware/AddCacheKey.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Rogue\Http\Middleware;
+
+use Closure;
+
+class AddCacheKey
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next, $type)
+    {
+        $params = $request->route()->parameters();
+        $post = $params['post'];
```

Comment:
Since this is so specific to this particular route, I feel like it might make more sense to inline [in the controller](https://github.com/DoSomething/rogue/blob/0297d021e46aa06f615b8de1a5f13ef55f2e9a38/app/Http/Controllers/ImagesController.php#L63):

```php
return $server->getImageResponse($post->getMediaPath(), $request->all())
  ->withHeader('Surrogate-Key', 'post-'.$post->id);
```

Reply:
It was getting a little messy in the controller because `$server->getImageResponse($post->getMediaPath(), $request->all())` returns a `StreamedResponse` instance, so I thought it would be nicer to work with it in the middleware. Also if we ever decided to cache/purge other things you can just use this middleware to do it.

Reply:
Oh, I made a bad assumption on what that class would be. If we wanted to keep it in the controller, it looks like this will work:

```php
$response = $server->getImageResponse($post->getMediaPath(), $request->all());
$response->headers->set('Surrogate-Key', 'post-'.$post->id);

return $response;
```

I'm not opposed to making a more generalized middleware here, but it feels a bit suspect to be pulling the `$post` out of the route parameters since that's tightly coupled to this use-case.

Reply:
otay! will update